core/consensus/src/engine.rs

```diff
@@ -599,6 +611,105 @@ impl<Adapter: ConsensusAdapter + 'static> ConsensusEngine<Adapter> {
         self.crypto.update(new_addr_pubkey_map);
         Ok(())
     }
+
+    async fn verify_proof(&self, ctx: Context, block: &FixedPill) -> ProtocolResult<()> {
+        if block.inner.block.header.height < 3 {
```

Comment:
Make sure the `header.prev_hash` is contiguous, and then delete the magic number.